shell/platform/windows/flutter_window.cc

```diff
@@ -0,0 +1,286 @@
+  event_data->phase = pointer_is_down_ ? FlutterPointerPhase::kMove
+                                       : FlutterPointerPhase::kHover;
+}
+
+void FlutterWindow::SendPointerMove(double x, double y) {
+  FlutterPointerEvent event = {};
+  event.x = x;
+  event.y = y;
+  SetEventPhaseFromCursorButtonState(&event);
+  SendPointerEventWithData(event);
+}
+
+void FlutterWindow::SendPointerDown(double x, double y) {
+  pointer_is_down_ = true;
+  FlutterPointerEvent event = {};
+  event.phase = FlutterPointerPhase::kDown;
+  event.x = x;
+  event.y = y;
+  SendPointerEventWithData(event);
+}
+
+void FlutterWindow::SendPointerUp(double x, double y) {
+  pointer_is_down_ = false;
+  FlutterPointerEvent event = {};
+  event.phase = FlutterPointerPhase::kUp;
+  event.x = x;
+  event.y = y;
+  SendPointerEventWithData(event);
+}
+
+void FlutterWindow::SendChar(unsigned int code_point) {
+  for (const auto& handler : keyboard_hook_handlers_) {
+    handler->CharHook(this, code_point);
+  }
+}
+
+void FlutterWindow::SendKey(int key, int scancode, int action, int mods) {
+  for (const auto& handler : keyboard_hook_handlers_) {
+    handler->KeyboardHook(this, key, scancode, action, mods);
+  }
+}
+
+void FlutterWindow::SendScroll(double delta_x, double delta_y) {
+  FlutterPointerEvent event = {};
+  SetEventLocationFromCursorPosition(&event);
+  SetEventPhaseFromCursorButtonState(&event);
+  event.signal_kind = FlutterPointerSignalKind::kFlutterPointerSignalKindScroll;
+  // TODO: See if this can be queried from the OS; this value is chosen
+  // arbitrarily to get something that feels reasonable.
+  const int kScrollOffsetMultiplier = 20;
+  event.scroll_delta_x = delta_x * kScrollOffsetMultiplier;
+  event.scroll_delta_y = delta_y * kScrollOffsetMultiplier;
+  SendPointerEventWithData(event);
+}
+
+void FlutterWindow::SendPointerEventWithData(
+    const FlutterPointerEvent& event_data) {
+  // If sending anything other than an add, and the pointer isn't already added,
+  // synthesize an add to satisfy Flutter's expectations about events.
+  if (!pointer_currently_added_ &&
+      event_data.phase != FlutterPointerPhase::kAdd) {
+    FlutterPointerEvent event = {};
+    event.phase = FlutterPointerPhase::kAdd;
+    event.x = event_data.x;
+    event.y = event_data.y;
+    SendPointerEventWithData(event);
+  }
+  // Don't double-add (e.g., if events are delivered out of order, so an add has
+  // already been synthesized).
+  if (pointer_currently_added_ &&
+      event_data.phase == FlutterPointerPhase::kAdd) {
+    return;
+  }
+
+  FlutterPointerEvent event = event_data;
+  // Set metadata that's always the same regardless of the event.
+  event.struct_size = sizeof(event);
+  event.timestamp =
+      std::chrono::duration_cast<std::chrono::microseconds>(
+          std::chrono::high_resolution_clock::now().time_since_epoch())
+          .count();
+
+  // Since the app is running in per-monitor V2 DPI mode, the application has a
```

Comment:
Can you clarify what this means for me/how this is set? This is a library, so I'm a bit concerned by a reference to "the app".